authorizer/secret.go

```diff
@@ -0,0 +1,142 @@
+package authorizer
+
+import (
+	"context"
+
+	"github.com/influxdata/influxdb"
+)
+
+var _ influxdb.SecretService = (*SecretService)(nil)
+
+// SecretService wraps a influxdb.SecretService and authorizes actions
+// against it appropriately.
+type SecretService struct {
+	s influxdb.SecretService
+}
+
+// NewSecretService constructs an instance of an authorizing secret serivce.
+func NewSecretService(s influxdb.SecretService) *SecretService {
+	return &SecretService{
+		s: s,
+	}
+}
+
+func newSecretPermission(a influxdb.Action, orgID influxdb.ID) (*influxdb.Permission, error) {
+	return influxdb.NewPermission(a, influxdb.SecretsResourceType, orgID)
+}
+
+func authorizeReadSecret(ctx context.Context, orgID influxdb.ID) error {
+	p, err := newSecretPermission(influxdb.ReadAction, orgID)
+	if err != nil {
+		return err
+	}
+
+	if err := IsAllowed(ctx, *p); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func authorizeWriteSecret(ctx context.Context, orgID influxdb.ID) error {
+	p, err := newSecretPermission(influxdb.WriteAction, orgID)
+	if err != nil {
+		return err
+	}
+
+	if err := IsAllowed(ctx, *p); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// LoadSecret checks to see if the authorizer on context has read access to the secret key provided.
+func (s *SecretService) LoadSecret(ctx context.Context, orgID influxdb.ID, key string) (string, error) {
+	secret, err := s.s.LoadSecret(ctx, orgID, key)
+	if err != nil {
+		return "", err
+	}
+
+	if err := authorizeReadSecret(ctx, orgID); err != nil {
```

Comment:
Can this come before the call to `s.s.LoadSecret` since we already have the orgID?